src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -77,6 +77,19 @@ def has_attributes(self):
         return self.attributes.exists()
 
 
+class CategoryQuerySet(MP_NodeQuerySet):
+    def browsable(self):
+        """
+        Excludes non-public categories
+        """
+        return self.filter(is_public=True)
```

Comment:
This is not sufficient, all subcategories of non-public categories must be filtered out as well. If not bad thing happen, see #3160

Reply:
fixed in 3de975c